.release-notes/3816.md

```diff
@@ -0,0 +1,3 @@
+## Cleanup and fixes for sockets on Windows
+
+Fixed a few infelicities in the Windows socket code, in particular that clients connecting to an invalid port would think they were successfully connected.
```

Comment:
I dont think this is very meaningful to an end user. It's basically "bug fix".


I think something like

```
## Address Windows socket errors

BLAH BLAH things fixed
```

with perhaps a description for users of what was wrong.

Reply:
I've updated them.